src/processor/graph-update-completion-monitor.service.ts

```diff
@@ -0,0 +1,119 @@
+    }
+
+    super.onApplicationShutdown(signal);
+  }
+
+  constructor(
+    cacheManager: ReconnectionCacheMgrService,
+    private readonly schedulerRegistry: SchedulerRegistry,
+    blockchainService: BlockchainService,
+    private readonly cacheService: ReconnectionCacheMgrService,
+  ) {
+    super(cacheManager, blockchainService, new Logger(GraphUpdateCompletionMonitorService.name));
+  }
+
+  async processCurrentBlock(currentBlockHash: BlockHash, currentBlockNumber: number): Promise<void> {
+    // Get set of tx hashes to monitor from cache
+    let pendingTxns = await this.cacheService.getAllPendingTxns();
+
+    const block = await this.blockchainService.getBlock(currentBlockHash);
+    const extrinsicIndices: [HexString, number][] = [];
+    block.block.extrinsics.forEach((extrinsic, index) => {
+      if (Object.keys(pendingTxns).some((txHash) => txHash === extrinsic.hash.toHex())) {
+        extrinsicIndices.push([extrinsic.hash.toHex(), index]);
+      }
+    });
+
+    if (extrinsicIndices.length > 0) {
+      const at = await this.blockchainService.apiPromise.at(currentBlockHash);
+      const epoch = (await at.query.capacity.currentEpoch()).toNumber();
+      let totalCapacityWithdrawn: bigint = 0n;
+      const events = (await at.query.system.events()).filter(({ phase }) => phase.isApplyExtrinsic && extrinsicIndices.some((index) => phase.asApplyExtrinsic.eq(index)));
+      const capacityAmounts: bigint[] = events
+        .filter(({ event }) => at.events.capacity.CapacityWithdrawn.is(event))
+        .map(({ event }) => (event as unknown as any).data.amount.toBigInt());
+      totalCapacityWithdrawn = capacityAmounts.reduce((prev, curr) => prev + curr, totalCapacityWithdrawn);
+
+      // eslint-disable-next-line no-restricted-syntax
+      for (const [txHash, txIndex] of extrinsicIndices) {
+        const extrinsicEvents = events.filter(({ phase }) => phase.isApplyExtrinsic && phase.asApplyExtrinsic.eq(txIndex));
+        const hasSuccess = extrinsicEvents.some(({ event }) => at.events.utility.BatchCompleted.is(event));
+        const failureEvent = extrinsicEvents.find(({ event }) => at.events.system.ExtrinsicFailed.is(event));
+
+        // eslint-disable-next-line no-await-in-loop
+        const txStatus = pendingTxns[txHash];
+
+        if (failureEvent && at.events.system.ExtrinsicFailed.is(failureEvent.event)) {
+          if (hasSuccess) {
+            this.logger.warn(`Events for tx ${txHash} include both success and failure ???`);
+          }
+          const { asModule: moduleThatErrored, registry } = failureEvent.event.data.dispatchError;
+          const moduleError = registry.findMetaError(moduleThatErrored);
+          txStatus.error = moduleError.method;
+          txStatus.status = 'failed';
+        } else if (hasSuccess) {
+          txStatus.status = 'success';
+        } else {
+          this.logger.error(`Tx hash ${txHash} found in block, but neither success nor failure`);
```

Comment:
This should still set the txStatus right?

Reply:
It is not possible as it is a passthrough to https://paritytech.github.io/polkadot-sdk/master/pallet_utility/pallet/enum.Call.html#variant.batch_all

There is a separate batch that would allow it, but this one is all or nothing